validphys2/src/validphys/core.py

```diff
@@ -266,83 +264,153 @@ def get_kinlabel_key(process_label):
+    def uncertainties(self):
+        """Loads the uncertainties when needed"""
+        if self._uncertainties is None:
+            unyaml = yaml.safe_load(self._uncertainties_file.read_text(encoding="utf-8"))
+            K_MODE = "Mode"
+            K_TREAT = "Treatment"
+            sys_dfs = defaultdict(list)
+            # Below, some abusing of pandas dataframes
+            for key, data in unyaml.items():
+                if key == "stat":
+                    lol = pd.concat({"": pd.DataFrame.from_records(data, index="index")}, names=[K_MODE], axis=1)
+                    lol.rename(columns={"value": key}, inplace=True)
+                    stat_df = pd.concat({"": lol}, axis=1, names=[K_TREAT])
+                else:
+                    tmp = pd.DataFrame.from_records(data["errors"], index="index")
+                    tmp.rename(columns={"value": key}, inplace=True)
+                    mode = data["mode"]
+                    if "CORR" in mode:
+                        tmp_df = pd.concat({"CORR": tmp}, axis=1, names=[K_MODE])
+                    else:
+                        tmp_df = pd.concat({"UNCORR": tmp}, axis=1, names=[K_MODE])
+                    if "ADD" in mode:
+                        sys_dfs["ADD"].append(tmp_df)
+                    else:
+                        sys_dfs["MULT"].append(tmp_df)  
+            all_sys = {k: pd.concat(i) for k,i in sys_dfs.items()}
+            sys_df = pd.concat(all_sys, axis=1, names=[K_TREAT])
+            unc_df = pd.concat({"stat": stat_df, "sys": sys_df}, axis=1, names=["Type"])
+
+            self._uncertainties = Uncertainties(unc_df)
+        return self._uncertainties
+
+    @property
+    def kinematics(self):
+        """Reads the kinematics of the dataset into a pandas dataframe"""
+        if self._kinematics is None:
+            kinyaml = yaml.safe_load(self._kinematics_file.read_text(encoding="utf-8"))
+            kin_data = []
+            keys = []
+            for key, data in kinyaml.items():
+                kin_data.append(pd.DataFrame.from_records(data, index="index"))
+                keys.append(key.replace("kin_", ""))
+            kin_df = pd.concat(kin_data, axis=1, keys=keys).swaplevel(0,1, axis=1).sort_values(1, axis="columns")
+
+            self._kinematics = Kinematics(kin_df)
+        return self._kinematics
 
     @property
     def nsys(self):
-        return self.metadata.nsys
+        return self.uncertainties.nsys
 
     @property
     def ndata(self):
-        return self.metadata.ndata
+        """Return the number of datapoints"""
+        return len(self.data)
 
     @property
     def process_type(self):
-        return self.metadata.process_type
-
-    @property
-    def metadata(self):
-        if self._metadata is None:
-            self._metadata = peek_commondata_metadata(self.datafile)
-        return self._metadata
+        """Return the number of systematic uncertainties"""
+        return self.metadata.get("proctype")
```

Comment:
I guess we need to discuss what to do with process types, as it is a bit ambiguous. On one hand there is "the string that identifies what kinematics mean" (which is roughly what we have in the current commondata files). We may or may not need that with more extensive metadata. Then there are random groupings that people pull out of their hat (e.g. the process type for thcovmat), which we do need to store somewhere.

cc @enocera